lms/djangoapps/experiments/utils.py

```diff
@@ -247,7 +250,7 @@ def get_experiment_user_metadata_context(course, user):
                         # program has 3 courses (A, B and C), and the user previously purchased a certificate for A.
                         # The user is enrolled in audit mode for B. The "left to purchase price" should be the price of
                         # B+C.
-                        non_audit_enrollments = [enrollment for enrollment in user_enrollments if enrollment not in
+                        non_audit_enrollments = [en for en in user_enrollments if en not in
```

Comment:
I'm pretty sure this fixes the bug. The variable in this list comprehension was almost certainly leaking out and clobbering the `enrollment` variable.